src/Uploader/EltsUploader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+namespace TYPO3\Darth\Uploader;
+
+/*
+ * This file is part of the TYPO3 project.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use TYPO3\Darth\Service\EltsPublishService;
+
+class EltsUploader implements UploaderInterface
+{
+    /**
+     * @var EltsPublishService
+     */
+    private $client;
+
+    public function __construct()
+    {
+        $this->client = new EltsPublishService(
+            getenv('ELTS_REMOTE_USER'),
+            getenv('ELTS_REMOTE_SERVER'),
+            rtrim(getenv('ELTS_REMOTE_LOCATION'), '/') . '/'
+        );
+    }
+
+    public function getUploadStartMessage(): string
+    {
+        $username = getenv('ELTS_REMOTE_USER');
+        $location = getenv('ELTS_REMOTE_LOCATION');
+        return sprintf('Publishing as %s to %s', $username, $location);
+    }
+
+    public function upload(string $file, string $blobName)
```

Comment:
let's move the PublishService into the Uploader, so the Uploader is not just a shell, but actually doing stuff. Or are there any other reasons for that extraction?

Reply:
We can do that